### PR TITLE
[PART 1] Make cached appeal attribute table joins a scope on tasks

### LINKED ISSUE #13914: Do not use assigned_to_location to sort and filter by assignee in org queues
When implementing back end sorting and filtering by assignee in #11780 for pagination, we noticed our [front end did not](https://github.com/department-of-veterans-affairs/caseflow/pull/11780/files#r314458380) sort or filter with the simple task assignee value. We opted to [match what the front end currently did and fix this later](https://dsva.slack.com/archives/CJL810329/p1565898494290900?thread_ts=1565893728.282400&cid=CJL810329). This is the ticket to fix this!

### AC
 - [ ] Back end sorts and filters based on the name (name for organization and full_name for user) of the assignee of the task

### PR DESCRIPTION
**Pull request by hschallhorn**
# THIS PR
Resolves #13914

PART 1 of the stack
PART 2 #14264
PART 3 #14265

### Description
Moves `joins(CachedAppeal.left_join_from_tasks_clause)` into a scope on tasks

### Acceptance Criteria
- [x] Tests pass (no changes)

# FULL STACK

### Description
Rather than displaying an appeal's problematic assigned_to_location, show the actual task assignee in task tables.

### Acceptance Criteria
- [ ] Task tables display the name of the org or the name of the user that is assigned the task
- [ ] Organization queues
   - [ ] Can sort by assignee name
   - [ ] Can filter by assignee name
   - [ ] Show the correct assignee filter options

#### Tasks assigned to users
1. Sign in as BVALSPORER
1. Navigate to the VLJ support queue http://localhost:3000/organizations/vlj-support
1. Confirm that assignee css_ids appear in the queue table
1. Confirm filter options look correct
1. Confirm filtering works
1. Confirm sorting works

#### Tasks assigned to organizations
1. Sign in as Jolly Postman
1. Navigate to any ama case (I searched for vet 321321321) and add 4 tasks (FOIA, AOD, change of addrress, and Congressional Interest)
1. Update all these tasks to be on hold
```ruby
uuid = ""
appeal = Appeal.find_by_uuid(uuid)
MailTask.where(appeal: appeal).each(&:on_hold!)
appeal.reload.treee
                                              ┌──────────────────────────────────────────────────────────────────────────────┐
Appeal 138 (direct_review) ────────────────── │ ID   │ STATUS  │ ASGN_BY       │ ASGN_TO           │ UPDATED_AT              │
└── RootTask                                  │ 352  │ on_hold │               │ Bva               │ 2020-05-07 16:00:44 UTC │
    ├── DistributionTask                      │ 353  │ on_hold │               │ Bva               │ 2020-05-13 00:46:14 UTC │
    │   ├── CongressionalInterestMailTask     │ 2623 │ on_hold │               │ MailTeam          │ 2020-05-13 00:46:14 UTC │
    │   │   └── CongressionalInterestMailTask │ 2624 │ on_hold │ JOLLY_POSTMAN │ LitigationSupport │ 2020-05-13 00:47:38 UTC │
    │   └── FoiaRequestMailTask               │ 2625 │ on_hold │               │ MailTeam          │ 2020-05-13 00:46:22 UTC │
    │       └── FoiaRequestMailTask           │ 2626 │ on_hold │ JOLLY_POSTMAN │ PrivacyTeam       │ 2020-05-13 00:47:38 UTC │
    ├── AodMotionMailTask                     │ 2627 │ on_hold │               │ MailTeam          │ 2020-05-13 00:46:35 UTC │
    │   └── AodMotionMailTask                 │ 2628 │ on_hold │ JOLLY_POSTMAN │ AodTeam           │ 2020-05-13 00:47:38 UTC │
    └── AddressChangeMailTask                 │ 2629 │ on_hold │               │ MailTeam          │ 2020-05-13 00:47:00 UTC │
        └── AddressChangeMailTask             │ 2630 │ on_hold │ JOLLY_POSTMAN │ Colocated         │ 2020-05-13 00:47:00 UTC │
            └── AddressChangeMailTask         │ 2631 │ on_hold │ JOLLY_POSTMAN │ BVALSPORER        │ 2020-05-13 00:47:38 UTC │
                                              └──────────────────────────────────────────────────────────────────────────────┘
```
1. Go to the mail queue's on hold tab http://localhost:3000/organizations/mail?tab=on_hold
1. Confirm that assignee org names appear in the queue table
1. Confirm filter options look correct
1. Confirm filtering works
1. Confirm sorting works

### User Facing Changes
 - [x] Queues show the actual assignees of each task (no one assignee for all tasks for one case)

 BEFORE|AFTER
 ---|---
Screen Shot 2020-05-12 at 8 53 39 PM|Screen Shot 2020-05-12 at 8 52 41 PM

 - [x] Queues show the css_id of the user assignee, rather than a mix of user id or css_id?

 BEFORE|AFTER
 ---|---
Screen Shot 2020-05-12 at 8 55 59 PM|Screen Shot 2020-05-15 at 1 25 39 PM

 - [x] Queues do not show "Case Storage" unless they are actually assigned to BVA

 BEFORE|AFTER
 ---|---
Screen Shot 2020-05-12 at 9 00 31 PM|Screen Shot 2020-05-15 at 5 02 39 PM

### CODE REVIEW

File: app/models/queue_column.rb [hunk 1]
```diff
@@ -108,8 +106,7 @@ def task_type_options(tasks)
   end
 
   def assignee_options(tasks)
-    tasks.joins(CachedAppeal.left_join_from_tasks_clause)
-      .group(:assignee_label).count.each_pair.map do |option, count|
+    tasks.with_cached_appeals.group(:assignee_label).count.each_pair.map do |option, count|
       label = self.class.format_option_label(option, count)
       self.class.filter_option_hash(option, label)
     end
```
Comment (lomky) [inline app/models/queue_column.rb:112]: Optional: Coverage of this method is missing from the spec! Would be nice to add.

File: app/models/task_filter.rb [hunk 2]
```diff
@@ -67,7 +67,7 @@ def initialize(args)
   end
 
   def filtered_tasks
-    where_clause.empty? ? tasks : tasks.joins(CachedAppeal.left_join_from_tasks_clause).where(*where_clause)
+    where_clause.empty? ? tasks : tasks.with_cached_appeals.where(*where_clause)
   end
```
Comment (lomky) [inline app/models/task_filter.rb:71]: Out of scope, but nice if you came across the answer: Why don't we ask for `.with_cached_appeals` on an empty where clause? Can we add a comment?

File: app/models/queue_column.rb (hunk 1 above)
Comment (hschallhorn) [reply app/models/queue_column.rb:112]: Good eye! This is added in the [next pr](https://github.com/department-of-veterans-affairs/caseflow/pull/14264/files#diff-12942c537488326df41fe58d7d41ae56R207) where we update how the assignee options are created

File: app/models/task_filter.rb (hunk 2 above)
Comment (hschallhorn) [reply app/models/task_filter.rb:71]: If the where clause is empty, we're not applying any filters, so we don't need to look at any associated cached appeal attributes, so theres no need to join with that table!

**Pull request merged**